charon/src/transform/remove_read_discriminant.rs

```diff
@@ -136,7 +136,8 @@ impl Transform {
                         }
                         _ => {
                             // The discriminant read is not followed by a `SwitchInt`. This can happen
-                            // in optimized MIR.
+                            // in optimized MIR. We replace `_x = Discr(_y)` with `match _y { 0 => { _x
+                            // = 0 }, 1 => { _x = 1; }, .. }`.
```

Comment:
You're adding back a sentence I removed; that's probably a merge mistake. Please re-read your PR before submitting to avoid such things.